components/timezone/src/types.rs

```diff
@@ -20,8 +20,12 @@ impl Default for GmtOffset {
 
 fn try_get_time_component(chars: &[u8]) -> Option<i32> {
     const DIGIT_0: i32 = b'0' as i32;
-    if chars.len() == 2 && chars[0].is_ascii_digit() && chars[1].is_ascii_digit() {
-        Some((chars[0] as i32 - DIGIT_0) * 10 + chars[1] as i32 - DIGIT_0)
+    if chars.len() == 2 && chars.iter().all(|c| c.is_ascii_digit()) {
+        Some(
+            chars
+                .iter()
+                .fold(0i32, |val, c| *c as i32 - DIGIT_0 + val * 10),
```

Comment:
hmm, i'm not sure if this fold is clearer. @robertbastian, thoughts?

Reply:
Food for thought. atoi is used as a dev dependency that can be used. But it's an additional dependency. Or you are fine with some more adhoc implementation then these can be cleaned. I am happy to adapt it to whatever you decide.

Reply:
(here it's kinda fine to use raw indexing since the length check is right there)

Reply:
Yeah, I get that. I was not happy when I found out clippy asks for 4 expects because of raw indexing. Oof. I am fine with that raw slicing. Anything else you want me to change?

Reply:
You can pattern match:

```rust
if let &[tens, ones] = chars {
  if tens.is_ascii_digit() && ones.is_ascii_digit() {
    Some((tens as i32 - DIGIT_0) * 10 + ones as i32 - DIGIT_0)
  } else {
    None
  }
} else {
  None 
}
```

However I'm not a fan of the ascii arithmetic. Consider using
```rust
if let &[tens, ones] = chars {
  Some((tens as char).to_digit(10)? * 10 + (ones as char).to_digit(10)?)
} else {
  None 
}
```

With my suggestion below the else case will go away as well.

Reply:
oh yeah forgot that was stable now!

Reply:
It looks quite good.